test/test_logging_console_output_handler.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <string>
+#include <vector>
+
+#include "osrf_testing_tools_cpp/scope_exit.hpp"
+#include "rcutils/logging.h"
+
+static void call_handler(
+  const rcutils_log_location_t * location,
+  int severity, const char * name, rcutils_time_point_value_t timestamp,
+  const char * format, ...)
+{
+  va_list args;
+  va_start(args, format);
+  rcutils_logging_console_output_handler(location, severity, name, timestamp, format, &args);
+  va_end(args);
+}
+
+// There are no outputs of the handler function, and the only result are fprintf() calls.
+// This is just a smoke test to check that the code can handle simple inputs cleanly.
+TEST(TestLoggingConsoleOutputHandler, typical_inputs) {
```

Comment:
@brawner meta: thinking again about this test, you _could_ mock `stdout` and/or `stderr` for the duration of the test. Won't block on it, but it is doable.

Reply:
I tried this out and couldn't get it to work. Assigning a new file pointer to stdout/stderr left me with blank files. Alternatively, using `freopen()` worked but it wouldn't let me reset stdout/stderr back in an easily portable way.

I like the idea of checking expected input on stdout/stderr, but I'll need a better solution, which may have to come later.

Reply:
Hmm, interesting. Oh well, it can be done in a follow-up PR.